### PR TITLE
Bez3 sim rebased

### PR DESCRIPTION
**Pull request by acrylic-origami**
- [ ] Only the minimal change is included in this PR the commits are kept minimal
- [ ] Code is readable and documented in the code
- [ ] Minimal tests have been added

### CODE REVIEW

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 1]
```diff
@@ -19,58 +20,58 @@
...
-    RIGHT_LEG_2 = 11
-    RIGHT_LEG_3 = 12
-    RIGHT_LEG_4 = 13
-    RIGHT_LEG_5 = 14
-    RIGHT_LEG_6 = 15
-    HEAD_1 = 16
-    HEAD_2 = 17
-    HEAD_CAMERA = 18
-    IMU = 19
+    TORSO = rospy.get_param("~joint_indices/TORSO", 1)
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:46]: Mistake

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 2]
```diff
@@ -244,6 +246,8 @@ def inverseKinematicsRightFoot(self, transformation):
         :return: Motor angles for the right foot
         """
         transformation[0:3, 3] = transformation[0:3, 3] - self.torso_to_right_hip[0:3, 3]
+        # transformation = np.array(transformation)[[1, 0, 2, 3],:]
+        # transformation = transformation[:,[1, 0, 2, 3]]
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:250]: Remove comments

File: soccer_pycontrol/src/soccer_pycontrol/test_walking.py [hunk 3]
```diff
@@ -54,11 +74,36 @@ def tearDown(self) -> None:
...
+        for _ in range(100):
+            if _ % 16 == 0:
+                _ = _
+            pb.stepSimulation()
+
+        pass
+
     @timeout_decorator.timeout(TEST_TIMEOUT)
     def test_walk_1(self):
+        # pb.setGravity(0, 0, -4)
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/test_walking.py:103]: remove comment

File: soccer_pycontrol/src/soccer_pycontrol/test_walking.py [hunk 4]
```diff
@@ -54,11 +74,36 @@ def tearDown(self) -> None:
         super().tearDown()
         del self.walker
 
+    @timeout_decorator.timeout(TEST_TIMEOUT)
+    def test_ik_1(self):
+        # t = self.walker.soccerbot.right_foot_init_position # Transformation((0, 0, 0.4), (0,0,0,1))
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/test_walking.py:79]: remove comment

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 5]
```diff
@@ -79,13 +80,14 @@ def __init__(self, pose, useFixedBase=False, useCalibration=False):
 
         home = expanduser("~")
         self.body = pb.loadURDF(
-            home + "/catkin_ws/src/soccerbot/bez1_description/urdf/soccerbot.urdf",
+            home
+            + f"/catkin_ws/src/soccerbot/{rospy.get_param('~robot_model', 'bez1')}_description/urdf/{rospy.get_param('~robot_model', 'bez1')}.urdf",
             useFixedBase=useFixedBase,
-            flags=pb.URDF_USE_INERTIA_FROM_FILE,
-            basePosition=[pose.get_position()[0], pose.get_position()[1], Soccerbot.standing_hip_height],
+            flags=pb.URDF_USE_INERTIA_FROM_FILE | (pb.URDF_MERGE_FIXED_LINKS if rospy.get_param("~_merge_fixed_links", False) else 0),
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:86]: why does this param have _ in from, change it to normal

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py [hunk 6]
```diff
@@ -79,13 +80,14 @@ def __init__(self, pose, useFixedBase=False, useCalibration=False):
...
+            basePosition=[pose.get_position()[0], pose.get_position()[1], Soccerbot.torso_height],
             baseOrientation=pose.get_orientation(),
         )
-
+        self.pybullet_offset = pb.getBasePositionAndOrientation(self.body)[0][:2] + (0,)  # pb.getLinkState(self.body, Links.TORSO)[4:6]
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:90]: what is (0,)?

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot_controller_ros.py [hunk 7]
```diff
@@ -112,7 +113,6 @@ def goal_callback(self, pose: PoseStamped):
             try:
                 self.t_new_path = self.new_path.dynamicallyUpdateGoalPosition(self.t, goal_position)
             except Exception as ex:
-                print(ex)
                 return
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot_controller_ros.py:116]: add ex print back

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot_ros.py [hunk 8]
```diff
@@ -79,7 +79,7 @@ def publishAngles(self):
             js.position.append(self.get_angles()[i])
         try:
             rospy.loginfo_once("Started Publishing Motors")
-            self.pub_all_motor.publish(js)
+            # self.pub_all_motor.publish(js)
```
Comment (Vuwij) [inline soccer_pycontrol/src/soccer_pycontrol/soccerbot_ros.py:82]: why is this commented out?

File: soccer_pycontrol/src/soccer_pycontrol/soccerbot.py (hunk 6 above)
Comment (acrylic-origami) [reply soccer_pycontrol/src/soccer_pycontrol/soccerbot.py:90]: Single element tuple containing 0. (0,)+(1,) = (0,1)

**Review by Vuwij — APPROVED** [review]

**Pull request merged**